Crypto/AsymmetricCrypto.cpp

```diff
@@ -0,0 +1,329 @@
+// This is an open source non-commercial project. Dear PVS-Studio, please check it.
+
+// PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
+
+#include "pch.h"
+#include "AsymmetricCrypto.h"
+
+AsymmetricCrypto::AsymmetricCrypto()
+{
+    m_keypair = std::make_shared<EVP_PKEY*>();
+    m_rsa_encr_ctx = std::make_shared<EVP_CIPHER_CTX*>();
+    m_rsa_decr_ctx = std::make_shared<EVP_CIPHER_CTX*>();
+
+    InitializeContext();
+
+    GenerateRsaKeypair();
+}
+
+AsymmetricCrypto::~AsymmetricCrypto()
+{
+    DestroyContext();
+}
+
+bool AsymmetricCrypto::InitializeContext()
+{
+    *m_rsa_encr_ctx = EVP_CIPHER_CTX_new();
+    *m_rsa_decr_ctx = EVP_CIPHER_CTX_new();
+
+    if (!*m_rsa_encr_ctx || !*m_rsa_decr_ctx) {
```

Comment:
If *m_rsa_decr_ctx will be nullptr - then you are not deleting *m_rsa_encr_ctx object, that already exist.